### PR TITLE
Add option to print the children of trusted steps in LFSC

### PR DESCRIPTION
**Pull request by ajreynol**
Also does minor refactoring for printing identifiers.

### CODE REVIEW

File: src/proof/lfsc/lfsc_printer.h [hunk 1]
```diff
@@ -139,6 +140,15 @@ class LfscPrinter
                           const LetBinding& lbind,
                           const std::map<const ProofNode*, size_t>& pletMap,
                           std::map<Node, size_t>& passumeMap);
+  /**
+   */
```
Comment (abdoo8080) [inline src/proof/lfsc/lfsc_printer.h:144]: ???

File: src/proof/lfsc/lfsc_printer.h [hunk 2]
```diff
@@ -41,10 +42,10 @@ class LfscPrintChannel;
  * It expects to print proof nodes that have been processed by the LFSC
  * proof post processor.
  */
-class LfscPrinter
+class LfscPrinter : protected EnvObj
 {
  public:
-  LfscPrinter(LfscNodeConverter& ltp);
+  LfscPrinter(Env& env, LfscNodeConverter& ltp);
```
Comment (abdoo8080) [inline src/proof/lfsc/lfsc_printer.h:48]: Are we using the environment anywhere?

File: src/options/proof_options.toml [hunk 3]
```diff
@@ -139,3 +139,11 @@ name   = "Proof"
...
   default    = "false"
   help       = "Check proof steps for satisfiability, for refutation soundness testing. Note this checks only steps for non-core proof rules"
+
+[[option]]
+  name       = "lfscExpandTrust"
+  category   = "expert"
+  long       = "lfsc-expand-trust"
+  type       = "bool"
+  default    = "false"
+  help       = "Print the children of trusted proof steps"
```
Comment (abdoo8080) [inline src/options/proof_options.toml:149]: Do we need this option? Can't we always print the children of trusted proof steps?

**Review by abdoo8080 — CHANGES_REQUESTED** [review]
I don't know what the motivation for this change is, but it looks good to me?

File: src/proof/lfsc/lfsc_printer.h (hunk 2 above)
Comment (ajreynol) [reply src/proof/lfsc/lfsc_printer.h:48]: Yes, to access options.

File: src/options/proof_options.toml (hunk 3 above)
Comment (ajreynol) [reply src/options/proof_options.toml:149]: There is a tradeoff, they are unnecessary for proof checking, but they ensure that the entire proof is reflected in the output.

**Comment by ajreynol** [comment]
The motivation for this change is to ensure the entire proof is printed.  Otherwise, each step that does not have a supported translation into LFSC will be a leaf (and its children will be discarded).

**Pull request merged**